### PR TITLE
PAAS-2131 bump to SAM 2.2.0 for jahia 8.1.2+

### PR DESCRIPTION
**Pull request by laurentfufu**
JIRA issue: https://jira.jahia.org/browse/PAAS-2131

Short description:

### CODE REVIEW

File: packages/jahia/consistency-check.yml [hunk 1]
```diff
@@ -157,8 +157,16 @@ actions:
...
+                    - isVersionStrictlyLower:
+                        a: ${globals.jahiaVersion}
+                        b: 8.1.2.0
+                        res: isLowerThan8120
+                    - if( ${globals.isLowerThan8120} ):
+                        - set:
+                            expectedVersion: 2.1.0
+                    - else:
+                        - set:
+                            expectedVersion: 2.2.0
```
Comment (amitc-jahia) [inline packages/jahia/consistency-check.yml:169]: Please update the action description to reflect correct SAM version as per Jahia version

File: packages/jahia/migrations/migrate-to-v17.yaml [hunk 2]
```diff
@@ -216,3 +217,20 @@ actions:
...
+    - if (${globals.isHigherOrEqualThan8120}):
+        - checkModule:
+            moduleSymname: server-availability-manager
+        - if ("${globals.runningVersion}" != "2.2.0"):
+            - disableHaproxyHealthcheck
+            - installOrUpgradeModule:
+                moduleSymname: server-availability-manager
+                moduleVersion: 2.2.0
+            - checkJahiaHealth
+            - enableHaproxyHealthcheck
```
Comment (amitc-jahia) [inline packages/jahia/migrations/migrate-to-v17.yaml:236]: New migration package (v18) is needed for this as the release is already prepared.

**Review by amitc-jahia — CHANGES_REQUESTED** [review]

File: packages/jahia/consistency-check.yml (hunk 1 above)
Comment (laurentfufu) [reply packages/jahia/consistency-check.yml:170]: done !

File: packages/jahia/migrations/migrate-to-v17.yaml (hunk 2 above)
Comment (laurentfufu) [reply packages/jahia/migrations/migrate-to-v17.yaml]: indeed, I fall into the trap every time :upside_down_face:

Comment (kevinvandel) [reply packages/jahia/migrations/migrate-to-v17.yaml]: Fool me once, shame on you ; fool me twice, shame on me :trollface:

**Review by kevinvandel — DISMISSED** [review]

File: packages/jahia/migrations/migrate-to-v18.yaml [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+  ### End Pre-checks
+
+  - upgradeSam                        # PAAS-2131
+
+  # Actions that require a restart:
+
+  # Restart action:
+  - if ("${globals.jahiaRollingRestartNeeded}" == "true"):
+      - install:
+          jps: "${globals.repoRootUrl}/packages/jahia/jahia-rolling-restart.yml"
```
Comment (kevinvandel) [inline packages/jahia/migrations/migrate-to-v18.yaml:42]: This can be removed

File: packages/jahia/migrations/migrate-to-v18.yaml [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+  - "../../../mixins/common.yml"
+  - "../../../mixins/jahia.yml"
+  - "../../../mixins/mariadb.yml"
+  - "../../../mixins/haproxy.yml"
+  - "../../../mixins/jcustomer.yml"
+  - "../../../mixins/elasticsearch.yml"
+
+globals:
+  version: 18
+  jahiaRollingRestartNeeded: "false"
```
Comment (kevinvandel) [inline packages/jahia/migrations/migrate-to-v18.yaml:18]: This can be removed

**Review by kevinvandel — CHANGES_REQUESTED** [review]

File: packages/jahia/migrations/migrate-to-v18.yaml (hunk 4 above)
Comment (laurentfufu) [reply packages/jahia/migrations/migrate-to-v18.yaml]: done

**Review by kevinvandel — APPROVED** [review]

**Pull request merged**